metadata/src/main/java/org/apache/kafka/controller/ProducerIdControlManager.java

```diff
@@ -34,10 +35,15 @@ public class ProducerIdControlManager {
 
     private final ClusterControlManager clusterControlManager;
     private final TimelineLong nextProducerId; // Initializes to 0
+    private final TimelineInteger brokerId;
```

Comment:
Since these are all changed at the same time, it would be better to have a `TimelineObject<LatestProducerIdBlock>`. Then you could just have a a `long, int, long` or whatever.